### ISSUE TITLE
Changing pins breaks rp2040 controller

### ISSUE THREAD

**Issue by micahvdm** [StonedEdge/Retro-Lite-CM5#31]
Let me start by saying this is a fantastic project! I’m reusing the rp2040 Gamepad code for my own device, but when I change the pins, then some of the buttons stop working…in particular the R3 button. Are there any other references where the button pins are set that could cause this?

**Comment by StonedEdge** [comment]
I will leave this one to @ginkage. Currently we're investigating the R3 shenanigans actually, as I also too had some issues with it freezing the controller with some new firmware.

**Comment by ginkage** [comment]
@micahvdm The pins are defined here: https://github.com/StonedEdge/Retro-Lite-CM5/blob/main/rp2040_gamepad/src/gamepad.cpp#L8-L36 and it's the only place, really.
The only reason R3 could misbehave in that code is when HotkeyPlus is pressed at the same time, other than that it's kind of unlikely. What pin are you trying to use, maybe it's mapped to some other function by default (e.g. UART or I2C or something)?

**Comment by micahvdm** [comment]
@ginkage I've tried with the default pins as well as with pins that are not mapped to anything else (i2c/spi/uart). The issue happens before I press any buttons. I plugged into a windows machine and all the buttons register except R3. Wiring is good because I wrote a circuitpython code with the same pins and all the buttons work. Not sure why it’s not working…I’ve tried to compile on WSL, an ubuntu machine and a vm and all end with the same result…thought I’d post here as I’ve spent about 4days troubleshooting and still can’t find what going on…baffled is about where I’m at now
